### PR TITLE
build: migrate to pnpm

### PR DESCRIPTION
**Pull request by spaenleh**
- **fix: move to pnpm**
- **fix: udpate for pnpm**

### CODE REVIEW

**Review by pyphilia — DISMISSED** [review]
LGTM, I just have a few comments!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ name: CI checks
 
 on:
   push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (spaenleh) [reply .github/workflows/ci.yml:6]: This is necessary to remove the double triggering of the action in the CI. 
`pull_request` event already takes care of running on every change to a pull-request (i.e.: rebasing, pushing a new commit etc.)

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "files.trimTrailingWhitespace": true
```
[the remark this answers is not in the document]
Comment (spaenleh) [reply .vscode/settings.json:2]: I think it makes sens to have this to share settings accross developers on the project. 
Of course We should not have too personal settings here, but there is an example of setting some options for eslint that I think makes sens to have in the folder settings for everyone.

**Review by ReidyT — APPROVED** [review]
Really nice!! LGTM 🤖

**Pull request merged**